src/Components/SmartComponents/Modals/OsExposureReportModal/OsExposureReportModal.js

```diff
@@ -0,0 +1,224 @@
+import React, { useEffect, useState } from 'react';
+import propTypes from 'prop-types';
+import {
+    Modal,
+    Button,
+    Form,
+    FormGroup,
+    TextInput,
+    Flex,
+    FlexItem,
+    Checkbox,
+    TextArea,
+    Text
+} from '@patternfly/react-core';
+import LifecycleSelect from './LifecycleSelect';
+import isEqual from 'lodash/isEqual';
+import {
+    getCveReportFilters,
+    VULNERABILITY_EXPOSURE_BY_OS_REPORT_TITLE
+} from '../../../../Helpers/constants';
+import { sortBy, xorWith } from 'lodash';
+
+const OsExposureReportModal = ({
+    isOsExposureModalOpen,
+    setOsExposureModalOpen,
+    osExposureData
+}) => {
+    const [reportTitle, setReportTitle] = useState(VULNERABILITY_EXPOSURE_BY_OS_REPORT_TITLE);
+    const [exposureMap, setExposureMap] = useState({});
+    const [selectedOsVersions, setSelectedOsVersions] = useState([]);
+    const [filterData, setFilterData] = useState({
+        impact: [],
+        advisory_available: ['true']
+    });
+    const [userNote, handleUserNoteChange] = useState('');
+    const CVE_REPORT_FILTERS = getCveReportFilters(false, true);
+    const ACTIVE_FILTERS = [
+        'impact',
+        'advisory_available'
+    ];
+
+    const handleModalClose = () => {
+        setOsExposureModalOpen(false);
+        setReportTitle(VULNERABILITY_EXPOSURE_BY_OS_REPORT_TITLE);
+        setExposureMap({});
+        setSelectedOsVersions([]);
+        handleUserNoteChange('');
+    };
+
+    useEffect(() => {
+        if (osExposureData?.length) {
```

Comment:
nit-pick: Should we add some empty state for the case when the exposure data is fetched and it's of 0 length (= no exposure data available)?

Reply:
There will never be a length of 0 because this is a static report. It shows all CVEs for all RHEL OS versions, whether a user has a system with that OS version or CVE.